File: frontend/src/services/patient.ts
import api from './api';

export interface Patient {
  id: string;
  name: string;
  phone: string;
  email?: string;
  telegramChatId?: number;
}

export interface PatientPayload {
  name: string;
  phone: string;
  email?: string;
}

export const patientService = {
  getAll: async (): Promise<Patient[]> => {
    const response = await api.get<Patient[]>('/patients');
    return response.data;
  },

  getById: async (id: string): Promise<Patient> => {
    const response = await api.get<Patient>(`/patients/${id}`);
    return response.data;
  },

  create: async (payload: PatientPayload): Promise<Patient> => {
    const response = await api.post<Patient>('/patients', payload);
    return response.data;
  },

  update: async (id: string, payload: PatientPayload): Promise<Patient> => {
    const response = await api.put<Patient>(`/patients/${id}`, payload);
    return response.data;
  },

  delete: async (id: string): Promise<void> => {
    await api.delete(`/patients/${id}`);
  },
};

File: backend/src/main/java/com/omnib2b/api/domain/Patient.java
package com.omnib2b.api.domain;

import com.omnib2b.api.core.entity.BaseTenantEntity;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import java.util.UUID;

@Data
@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "patients")
public class Patient extends BaseTenantEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String phone;

    private String email;

    @Column(name = "telegram_chat_id")
    private Long telegramChatId;
}

